improver/metadata/check_datatypes.py

```diff
@@ -30,201 +30,85 @@
-    """
-    if item.units.is_time_reference():
-        return TIME_REFERENCE_DTYPE
-    if issubclass(item.dtype.type, np.integer):
-        return np.int32
-    return np.float32
-
-
-def check_datatypes(cube, coords=None):
-    """
-    Function to check the datatypes of cube diagnostics and coordinates
-    against the expected standard.  The default datatype is float32;
-    integer quantities are expected to be 32-bit with the exception of
-    absolute time.
-
-    Args:
-        cube (iris.cube.Cube):
-            Cube to be checked
-        coords (list or None):
-            List of coordinate names to check.  If None, checks all
-            coordinates present on the input cube.
-
-    Raises:
-        ValueError: if the input cube does not conform to the datatypes
-            standard
-    """
-    # construct a list of objects (cube and coordinates) to be checked
-    object_list = _construct_object_list(cube, coords)
-
-    msg = ('{} datatype {} does not conform to expected standard ({})\n')
-    error_string = ''
-    for item in object_list:
-        # allow string-type objects
-        if item.dtype.type == np.unicode_:
-            continue
-
-        # check numerical datatypes
-        required_dtype = _get_required_datatype(item)
-        if item.dtype.type != required_dtype:
-            error_string += msg.format(item.name(), item.dtype, required_dtype)
-
-        if (hasattr(item, "bounds") and item.bounds is not None
-                and item.bounds.dtype.type != required_dtype):
-            error_string += msg.format(
-                item.name()+' bounds', item.bounds.dtype, required_dtype)
-
-    # if any data was non-compliant, raise details here
-    if error_string:
-        raise ValueError(error_string)
-
-
-def _check_units_and_dtype(obj, units, dtype):
-    """
-    Check whether the units and datatype of the input object conform
-    to the standard given.
-
-    Args:
-        obj (iris.cube.Cube or iris.coords.Coord):
-            Cube or coordinate to be checked
-        units (str):
-            Required units
-        dtype (type):
-            Required datatype
-
-    Returns:
-        bool:
-            True if object conforms; False if not
-    """
-    if Unit(obj.units) != Unit(units):
-        return False
-
-    if obj.dtype.type != dtype:
-        return False
 
+def check_dtype(obj):
+    # if coord, acts on coord.points
+    req_dtype = get_required_dtype(obj)
+    dtype_ok = obj.dtype == req_dtype
+
+    if isinstance(obj, iris.coords.Coord) and obj.has_bounds():
+        # check bounds - want the same dtype as the points
+        bounds_dtype_ok = obj.bounds.dtype == req_dtype
+        dtype_ok = dtype_ok and bounds_dtype_ok
+    return dtype_ok
+
+
+def get_required_units(obj):
+    if _is_time_coord(obj):
+        return TIME_COORDS[obj.name()].units
+    # No special requirement on non-time coordinate objects
+    return None
+
+
+def check_units(obj):
```

Comment:
Probably leave `check_datatypes` as the name of the below function, but rename this to `valid_units` (and `valid_dtype` above)?